rest/src/main/kotlin/com/gitlab/kordlib/rest/Image.kt

```diff
@@ -18,19 +18,42 @@ class Image private constructor(val data: ByteArray, val format: Format) {
         }
 
     companion object {
-        fun raw(data: ByteArray, format: Format): Image {
-            return Image(data, format)
+        fun raw(data: ByteArray, format: Format, size: Size): Image {
+            return Image(data, format, size)
         }
 
         suspend fun fromUrl(client: HttpClient, url: String): Image = with(Dispatchers.IO) {
             val call = client.request<HttpResponse>(url) { method = HttpMethod.Get }
+            val size = Size.fromResolution(call.request.url.parameters["size"])
             val contentType = call.headers["Content-Type"]
-                    ?: error("expected 'Content-Type' header in image request")
+                ?: error("expected 'Content-Type' header in image request")
 
             @Suppress("EXPERIMENTAL_API_USAGE")
             val bytes = call.content.toByteArray()
 
-            Image(bytes, Format.fromContentType(contentType))
+            Image(bytes, Format.fromContentType(contentType), size)
+        }
+    }
+
+    /**
+     * Represents size of the Image as specified by
+     */
+    enum class Size(val resolution: String) {
```

Comment:
Why represent resolution as a String if the values can only be integers?

Reply:
Hi, I thought the same at first 😅, but everytime we need to encode the url it may consume time because Int to String conversion isn't really cheap (as process goes through creating a byte buffer, then processing each digit to char values, and then copying the buffer to String).

Reply:
1) While I'm glad to hear you take performance to such a high standard, this sounds like a case of premature optimization. I doubt concatenating ints to strings for Images will be in anyone's hot loop when using Kord.

2) Your  explanation isn't correct. String concatenation in Kotlin is delegated to a `Stringbuilder`, which uses [Integer.getChars](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/Integer.java#L486-L518) to provide a zero-garbage implementation for adding numbers to its internal byte array.